lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1360,6 +1360,45 @@ LogicalResult SubaccessOp::canonicalize(SubaccessOp op,
       });
 }
 
+OpFoldResult MultibitMuxOp::fold(ArrayRef<Attribute> operands) {
+  // If there is only one input, just return it.
+  if (operands.size() == 2)
+    return getOperand(1);
+
+  if (auto constIndex = getConstant(operands[0])) {
+    auto index = constIndex->getExtValue();
+    // operands[0] is index so (value + 1) is the index we want.
+    if (0 <= index && static_cast<unsigned>(index) + 1 < operands.size())
```

Comment:
Super nit, but please put the dynamic thing you're testing on the LHS of a comparison and the bounds on the RHS, e.g. "index >= 0".

Are there any overflow conditions to watch out for here casting to unsigned and with getExtValue?

Reply:
> Are there any overflow conditions to watch out for here casting to unsigned and with getExtValue?

I think it is ok to cast to unsigned because subindex op also requires index to be unsigned 32-bit integer.